scene/resources/portable_compressed_texture.cpp

```diff
@@ -122,7 +138,7 @@ void PortableCompressedTexture2D::create_from_image(const Ref<Image> &p_image, C
 	Vector<uint8_t> buffer;
 
 	buffer.resize(20);
-	encode_uint32(p_compression_mode, buffer.ptrw());
+	encode_uint16(p_compression_mode, buffer.ptrw());
```

Comment:
Contrary to reasonable security practices, Vector::resize() is *not zeroed* in Godot C++. This is done to keep reviewers and contributors on their toes, and I certainly am standing on my toes as I write this.

While there are several cases below that write to these extra bytes, I'm having trouble following the code flow below. I would rather guarantee that something is written here by default, even if it is later overwritten in all reasonable cases.
```suggestion
	encode_uint16(p_compression_mode, buffer.ptrw());
	encode_uint16(CompressedTexture2D::DATA_FORMAT_UNDEFINED, buffer.ptrw() + 2);
```

Reply:
Looks good now.